api/v3/Activity/Getdayswithactivities.php

```diff
@@ -67,14 +72,23 @@ function civicrm_api3_activity_getdayswithactivities(array $params) {
 
   $query->groupBy('a.activity_date_time');
   $result = $query->execute()->fetchAll();
-
-  $uniqueDates = array_unique(array_map(function ($row) {
-    return explode(' ', $row['activity_date_time'])[0];
-  }, $result));
-
-  $params['sequential'] = 1;
-
-  return civicrm_api3_create_success($uniqueDates, $params, 'Activity', 'getdayswithactivities');
+  $data = [];
+  if ($shouldGroupByField) {
+    foreach ($result as $row) {
+      $activityDate = explode(' ', $row['activity_date_time'])[0];
+      if (isset($data[$row[$params['options']['group_by_field']]]) && array_search($activityDate, $data[$row[$params['options']['group_by_field']]]) !== FALSE) {
```

Comment:
can you move the condition out and split into multiple lines? This is not supposed to pass code style because of the line's length.

Reply:
Done, but it does not complain for lines length.